File: ch01/1.1/C/machine_epsilon_hat.c
#include <stdio.h>

int main() {
	float feps;
	feps = 1.0f;
	while ( feps + 1.0f > 1.0f ) {
		feps = feps * 0.5f;
	}

	printf("float machine epsilon hat = %e\n", feps);
	return 0;
}

File: ch01/1.1/Fortran/machine_epsilon_hat.f90
program find_epsilon
    implicit none
    real(kind=4) :: feps
    feps = 1.0e0
    do while (feps + 1.0e0 > 1.0e0)
        feps = feps * 0.5e0
    end do
    print *, 'Floating point machine epsilon is:', feps
end program find_epsilon
